app/src/main/java/com/android/voyageur/ui/search/SearchScreen.kt

```diff
@@ -418,19 +417,14 @@ fun UserSearchResultItem(
   val isConnected = connectionStatus == ConnectionState.Available
   val currentUser by userViewModel.user.collectAsState()
   val sentFriendRequests by userViewModel.sentFriendRequests.collectAsState()
-  // Determine if the user is the currently logged-in user
-  val isCurrentUser = currentUser?.id == user.id
+  val receivedFriendRequests by userViewModel.friendRequests.collectAsState()
 
-  val isContactAdded by
-      remember(currentUser, user) {
-        derivedStateOf { currentUser?.contacts?.contains(user.id) ?: false }
-      }
+  val isCurrentUser = currentUser?.id == user.id
+  val isContactAdded = currentUser?.contacts?.contains(user.id) ?: false
+  val isRequestPending = sentFriendRequests.any { it.to == user.id }
+  val isRequestReceived = receivedFriendRequests.any { it.from == user.id }
 
-  val isRequestPending by
-      remember(sentFriendRequests, user) {
-        derivedStateOf { sentFriendRequests.any { it.to == user.id } }
-      }
-  LaunchedEffect(isContactAdded, isRequestPending) {
+  LaunchedEffect(isContactAdded, isRequestPending, isRequestReceived) {
```

Comment:
before merging you should remove this log which I assume was used for debugging purposes

Reply:
Thank you for pointing that out! The log statement in the LaunchedEffect was indeed intended for debugging purposes to monitor recompositions during development. You’re absolutely right that it should be removed before merging to ensure cleaner production code.

Reply:
Thank you for pointing that out! The log statement in the LaunchedEffect was indeed intended for debugging purposes to monitor recompositions during development. You’re absolutely right that it should be removed before merging to ensure cleaner production code.